### ISSUE TITLE
Inference on Custom RS-only Data:

### ISSUE THREAD

**Issue by Sipher-bd** [ljzycmd/BSRSC#3]
Hello! First, thank you for the incredible work on AdaRSCNet and the BSRSC dataset.

I am trying to run inference using the pre-trained weights (adarsc_bsrsc.pth) on a custom, real-world video clip (downloaded from internet and some are made with drones ). Because I only have Rolling Shutter (RS) frames and no Global Shutter (GS) ground truth, I am encountering some domain gap issues related to the camera readout time and would love your advice on how to properly adapt the inference pipeline.

**My Setup & Hardware:**

GPU: NVIDIA RTX 5070 (12 GB VRAM)

Custom CUDA Modification: When initially running the code, I encountered a "no kernel image is available for execution on the device" error stemming from the custom corr_cuda.py module (likely due to the newer RTX 5070 architecture). I used an AI assistant to help me modify/recompile the CUDA files to get it running.

Data: 640 x 480 resolution, 30 FPS (extracted into 120 sequential frames).

Workaround used: I duplicated my RS frames into a dummy GS folder to allow the dataloader to run without crashing.

**The Problem:**

When I run test.py with the default adarsc_bsrsc.yaml config, the model runs perfectly (118 iterations for 120 frames), but it applies almost zero correction to the output.

CRITICAL DETAIL: The model and my modified corr_cuda.py file work perfectly when I test them on the official BSRSC dataset (giving excellent visual results and normal PSNR metrics). It is only failing to warp my custom, real-world footage

Because I am using my RS images as the fake GS ground truth, my PSNR is very high (32.14), which visually confirms the model is just outputting the distorted input image.

What I have tried so far:

1. Adjusting the temporal window: I changed interval: 1 to interval: 2 and interval: 3 in the config to artificially increase the camera velocity. This successfully dropped the PSNR to ~31.5, indicating a heavier warp was applied, but the visual results still do not accurately deskew the custom footage.

**terminal output:** 

(bsrsc_nightly) sipher@sipher:~/Desktop/code/SimDeblur/BSRSC$ CUDA_VISIBLE_DEVICES=0 python test.py configs/adarsc_bsrsc.yaml ./checkpoints/adarsc_bsrsc.pth 
Cannot inport EDVR modules!!!
Cannot import STFAN modules!!!
25 May 2026 15:26:51 INFO simdeblur: logs are saved to ./workdir/adarscnet/bsrsc/000_AdaRSCNet_lr2e4_BSRSC_e401_B1G4/tested/20260525_152651
/home/sipher/anaconda3/envs/bsrsc_nightly/lib/python3.10/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
/home/sipher/anaconda3/envs/bsrsc_nightly/lib/python3.10/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=VGG19_Weights.IMAGENET1K_V1`. You can also use `weights=VGG19_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
25 May 2026 15:26:52 INFO simdeblur: Loss items: 
25 May 2026 15:26:52 INFO simdeblur:     CharbonnierLoss, weight: 1.0
25 May 2026 15:26:52 INFO simdeblur:     PerceptualLossVGG19, weight: 0.01
25 May 2026 15:26:52 INFO simdeblur:     VariationLoss, weight: 0.001
25 May 2026 15:26:52 INFO simdeblur: Checkponit loaded to model successfully!
25 May 2026 15:26:52 INFO simdeblur: Using checkpoint loaded from ./checkpoints/adarsc_bsrsc.pth for testing.
validation on gpu0::   0%|                              | 0/118 [00:00<?, ?it/s]/home/sipher/anaconda3/envs/bsrsc_nightly/lib/python3.10/site-packages/torch/functional.py:505: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at /pytorch/aten/src/ATen/native/TensorShape.cpp:4381.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
validation on gpu0:: 100%|████████████████████| 118/118 [01:13<00:00,  1.60it/s]
Memory:  146916352
mean PSNR: 32.14  mean SSIM: 0.9282  total time: 64.72s  average time: 0.5485s FPS: 1.82

**updated corr_cuda.py(along with old code):**

[corr_cuda.py](https://github.com/user-attachments/files/28218163/corr_cuda.py)

**sample dataset :** 

https://github.com/user-attachments/assets/7c8643b6-10de-424c-9990-17ce1f878579

**output dataset :** 

https://github.com/user-attachments/assets/798f8cbf-b65e-4db1-919c-9043d908c04e

looking forward for your reply .
thanks for your time

**Comment by ljzycmd** [comment]
Hi @Sipher-bd, thanks a lot for your interest in this project, and sorry for the delayed reply.

I went through your description, and your diagnosis is very likely correct: your custom RS footage probably has different exposure time, readout time, and inter‑frame timing compared to the BSRSC training distribution. Since our model implicitly learns a motion–readout prior from BSRSC, a large mismatch in these timing parameters can cause the model to apply almost no correction.

I also noticed that you closed the issue, does that mean you’ve already found a solution on your side? If you have additional questions, feel free to contact me.

**Comment by Sipher-bd** [comment]
Hi @ljzycmd ,

Thank you so much for looking into this and providing such a clear explanation! Your diagnosis makes perfect sense—drone cameras often have vastly different sensor readout speeds and exposure profiles compared to standard handheld or automotive cameras, so the mismatch in the motion–readout prior explains why the model is bypassing the correction.

Regarding the closed issue: I closed it because I realised the lack of correction wasn't a bug in your code, but rather a fundamental data mismatch on my end. However, I would love to get your insights on how I might adapt or fine-tune your model to work effectively with this drone data.

If you have some time, I would highly appreciate your guidance on the following questions:

1. Fine-Tuning for Custom Drone Hardware
Since the timing parameters (exposure, readout, inter-frame) of my drone footage differ heavily from the BSRSC training distribution, what is the best approach to fine-tune the model?

Should I focus on fine-tuning the existing weights with a smaller set of drone data, or will the implicit prior require a full retraining?

Is there a way to explicitly inject or adapt the camera's timing parameters during training/inference?

2. Dynamic Scenes vs. Ego-Motion
Does the model primarily resolve rolling shutter artifacts caused by camera/ego-motion, or is it also capable of handling fast-moving dynamic objects within a relatively static frame? Drone footage frequently features both simultaneously.

3. Handling Complex Distortions (Jello Effect & Heat Haze)
Drone footage suffers from unique environmental distortions. How do you expect the model to perform against:

High-frequency motor vibrations (commonly known as the "jello effect"), which cause rapid, non-linear rolling shutter warping?

Atmospheric turbulence/heat distortion, which introduces stochastic warping?

4. Synthetic Data Generation (Bonus Question)
If I need to generate a custom synthetic dataset to match my drone's profile, do you have any recommendations or scripts from your BSRSC pipeline that I could reuse to simulate the correct readout and exposure times?

Thank you again for your time, responsiveness, and for sharing this amazing project with the community. Your insights will be incredibly valuable for my research/project!
